sdk/src/wallet/operations/syncing/addresses/output_ids/account_foundry.rs

```diff
@@ -77,13 +77,9 @@ impl<S: 'static + SecretManage> Wallet<S> {
             }
         }
 
-        let mut output_ids = HashSet::new();
         let results: Vec<Result<OutputIdsResponse, WalletError>> = futures::future::try_join_all(tasks).await?;
-
-        for res in results {
-            let foundry_output_ids = res?;
-            output_ids.extend(foundry_output_ids.items);
-        }
+        let responses: Vec<OutputIdsResponse> = results.into_iter().collect::<Result<Vec<_>, _>>()?;
+        let output_ids: HashSet<OutputId> = HashSet::from_iter(responses.into_iter().flat_map(|res| res.items));
```

Comment:
I know there was a HashSet also before, but this shouldn't be necessary as one Foundry can only be controlled be one account

Reply:
if you're absolutely sure I can remove the HashSet

Reply:
Node needs to return wrong information for it  to be be wrong, but that would be no different to the other functions where we get output ids